src/units/abilities.cpp

```diff
@@ -563,27 +563,41 @@ template std::pair<int, map_location> unit_ability_list::get_extremum<std::great
 
 namespace {
 
+    struct special_match
+    {
+        std::string tag_name;
+        config cfg;
```

Comment:
Shouldn't cfg be pointer, to reduce copying? I'm guessing it should be a `const config *` since the original code passed pointers-to-const around.